source/client/io/ModelReader.ts

```diff
@@ -100,28 +102,86 @@ export default class ModelReader
         return ModelReader.mimeTypes.indexOf(mimeType) >= 0;
     }
 
-    get(url: string): Promise<Object3D>
+    get(url: string, {signal}:{signal?:AbortSignal}={}): Promise<Object3D>
     {
-        return new Promise((resolve, reject) => {
-            this.gltfLoader.load(url, gltf => {
-                resolve(this.createModelGroup(gltf));
-            }, null, error => {
-                if(this.gltfLoader === null || this.gltfLoader.dracoLoader === null) {
-                    // HACK to avoid errors when component is removed while loading still in progress.
-                    // Remove once Three.js supports aborting requests again.
-                    resolve(null);
+        this.loadingManager.itemStart(url);
+        let resourcePath = LoaderUtils.extractUrlBase( url );
+        return this.loadModel(url, {signal})
+        .then(data=>this.gltfLoader.parseAsync(data, resourcePath))
+        .then(gltf=>this.createModelGroup(gltf))
+        .then((result)=>{
+            this.loadingManager.itemEnd(url);
+            return result;
+        }, (e)=> {
+            this.loadingManager.itemError(url);
```

Comment:
Could change this to `signal.aborted ? this.loadingManager.itemEnd(url) : this.loadingManager.itemError(url);` I *think* the only ramification here is the loaded file count.

Reply:
We could. Do you think the AbortError should be silenced? Maybe we should let it surface at least initially to help troubleshoot if it aborts when it shouldn't.

Reply:
Looks like proper use is to always call itemEnd even after an error

https://github.com/mrdoob/three.js/blob/dev/src/loaders/FileLoader.js#L261

Reply:
IMHO an AbortError is an abort without a reason. But I think it's fine how it is. There is still a little wonkiness with how we are handling the interaction prompt, but I think that's a separate issue that is just highlighted by this update.